hack/update-bazel.sh

```diff
@@ -27,6 +27,10 @@ TMP_GOPATH=$(mktemp -d)
 # ref: https://github.com/kubernetes/kubernetes/blob/4e2f5e2212b05a305435ef96f4b49dc0932e1264/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD#L23-L49
 rm -f ${TESTINFRA_ROOT}/vendor/k8s.io/apimachinery/pkg/util/sets/{BUILD,BUILD.bazel}
 
+# manually remove BUILD as testdata cannot be found
+rm -f ${TESTINFRA_ROOT}/vendor/k8s.io/client-go/util/cert/BUILD
```

Comment:
cc @BenTheElder @ixdy is this fine?

Reply:
in the future the publishing bot will start doing this last I saw

Reply:
hm, is gazelle not removing the defunct `go_test`?

Reply:
it might be now. Let me retry.

Reply:
no this is still needed

Reply:
```shell
$ bazel build //...            
INFO: Analysed 2077 targets (524 packages loaded).
INFO: Found 2077 targets...
ERROR: missing input file '//vendor/k8s.io/client-go/util/cert:testdata/dontUseThisKey.pem'
ERROR: /usr/local/google/home/sebastienvas/go/src/k8s.io/test-infra/boskos/BUILD.bazel:16:1: //boskos:go_default_test: missing input file '//vendor/k8s.io/client-go/util/cert:testdata/dontUseThisKey.pem'
ERROR: /usr/local/google/home/sebastienvas/go/src/k8s.io/test-infra/boskos/BUILD.bazel:16:1 1 input file(s) do not exist
INFO: Elapsed time: 6.559s, Critical Path: 0.36s
FAILED: Build did NOT complete successfully
```